packages/client/src/comp/util/Signature.tsx

```diff
@@ -24,25 +25,42 @@ type Props = React.PropsWithChildren<{
 
 export const Signature: React.FC<Props> = ({ setSignature, setSignatureType, signatureType }) => {
   const { query } = useAppHistory()
-  return <Margin>
-    <Switchable visibleSwitchButton={query['case'] !== 'upload'} signatureType={signatureType} setSignatureType={setSignatureType}>
-    {
-      (signatureType) => {
-        if (signatureType === 'canvas') {
-          return <div>
-            <div>
-              <Canvas setSignature={setSignature} width={width} height={height}/>
+  const { locale } = AddressContainer.useContainer()
```

Comment:
Check out Switchable.  Use `visibleSwitchButton` to disable the switchable button on top.  Then you only have to render `<Canvas>` and `<Upload>` within `< Switchable/>`

Reply:
Thanks, that looks way better!